tests/unit/moduleapi/basics.tcl

```diff
@@ -32,6 +32,32 @@ start_server {tags {"modules"}} {
     test "Unload the module - test" {
         assert_equal {OK} [r module unload test]
     }
+
+    test "Unload the module in multi - test" {
+        r module load $testmodule
+        r multi
+        r module unload test
+        r test.basics
+        assert_error {*Invalid command: test.basics*} {r exec}
+    }
+
+    test "Unload the module when module command already in multi queue - test" {
+        set rd [redis_client]
+
+        r module load $testmodule
+        r multi
+        r test.basics
+        assert_equal {OK} [$rd module unload test]
+        assert_error {*Invalid command: test.basics*} {r exec}
+    }
+
+    test "Unload the module when module command was referenced by client - test" {
+        set rd [redis_client]
+        r module load $testmodule
+        r test.basics
+        assert_equal {OK} [$rd module unload test]
+        assert_match "*cmd=test.basics*" [$rd client list]
+    }
```

Comment:
Maybe also test this: When test.basics is the last command of `r`, preventing the command from being freed, try to call test.basics again from another client (e.g. $rd).

This is not multi so this case is not covered by the check in execCommand, but I think it is already removed from the `server.commands`  dict so the command is not found.

Reply:
Done it.